lib/IO/String.pm

```diff
@@ -36,10 +36,48 @@ TODO
 
 #| An L<IO::Handle> implementation that writes to memory.
 class IO::String:ver<0.1.0>:auth<hoelzro> is IO::Handle {
-    has @.contents;
+    has Str $.buffer = '';
+    has Int $.pos = 0;
+
+    multi method open(IO::String:D:) {
+        my Str $new-buffer = '';
+        $!buffer := $new-buffer;
+        $!pos = 0;
+    }
+
+    multi method open(IO::String:D: Str $buffer is rw) {
+        $!buffer := $buffer;
```

Comment:
Yes. This is a feature the Perl 5 version of IO::String has and I think it's reasonable. On the other hand, it might be surprising, so maybe it should be available with a flag instead. Like:

```perl6
multi method open(IO::String:D: Str $buffer is rw, :bind!) { ... }
multi method open(IO::String:D: Str $buffer) { ... }
```